intera_interface/src/intera_interface/limb.py

```diff
@@ -519,3 +588,147 @@ def test_collision():
             raise_on_error=False,
             body=lambda: self.set_joint_positions(positions)
             )
+
+    def ik_request(self, pose,
+                   end_point='right_hand', joint_seed=None,
+                   nullspace_goal=None, nullspace_joint_names=None, nullspace_gain=0.4):
+        """
+        Inverse Kinematics request sent to IK Service
+
+        @type pose: geometry_msgs.Pose
+        @param pose: Cartesian pose of the end point
+        @type end_point: string
+        @param end_point: name of the end point (should be in URDF)
+        @type joint_seed: [double]
+        @param joint_seed: the joint angles for the initial IK guess (optional)
+        @type nullspace_goal: [double]
+        @param nullspace_goal: the desired joints to bias the solver (optional)
+        @type nullspace_joint_names: [string]
+        @param nullspace_joint_names: joint names for the nullspace_goal (optional)
+        @type nullspace_gain: double
+        @param nullspace_gain: gain used to bias toward the nullspace goal [0.0, 1.0] (optional)
+        @rtype: dict({str:float})
+        @return: valid joint positions if exists.  False if no solution is found.
+        """
+        # Add desired pose for inverse kinematics
+        ikreq = SolvePositionIKRequest()
+        hdr = Header(stamp=rospy.Time.now(), frame_id='base')
+        ikreq.pose_stamp.append(PoseStamped(header=hdr, pose=pose))
```

Comment:
We might need to guard against bad type inputs here... as you've documented, pose needs to be a `geometry_msgs.Pose`